diff --git a/src/streaming_load.cpp b/src/streaming_load.cpp
index ec5a189..9192e05 100644
--- a/src/streaming_load.cpp
+++ b/src/streaming_load.cpp
@@ -3,6 +3,7 @@
 #include <utility/assumptions.hpp>
 #include <utility/invariants.hpp>
 #include <boost/json.hpp>
+#include <boost/system/error_code.hpp>
 #include <unordered_map>
 #include <iostream>
 #include <sstream>
@@ -127,7 +128,7 @@ static void parse_instruction(Instruction& sala_instruction, boost::json::value
 
 std::istream& operator>>(std::istream& istr, Program& program)
 {
-    boost::json::error_code error_code;
+    boost::system::error_code error_code;
     boost::json::value const json_program{ boost::json::parse(std::string(std::istreambuf_iterator<char>(istr), {}), error_code) };
 
     boost::json::object const& root_obj = json_program.as_object();
